### PR TITLE
Refactor String.replaceAll() to Matcher.replaceAll()

### PR DESCRIPTION
**Pull request by Happy-Neko**
Fix for #596.

* Refactor `String.replaceAll(`) to `Matcher.replaceAll()`.

### CODE REVIEW

**Comment by Happy-Neko** [comment]
@davvd Please proceed with code review.

**Comment by davvd** [comment]
@happy-neko Many thanks, I will find someone to review it before we merge

**Comment by davvd** [comment]
@hdouss please review, thanks

File: src/main/java/org/takes/facets/fork/MediaType.java [hunk 1]
```diff
@@ -42,6 +43,11 @@
 final class MediaType implements Comparable<MediaType> {
 
     /**
+     * Media type priority clean pattern.
+     */
+    private static final Pattern CLEANER = Pattern.compile("[^0-9\\.]");
```
Comment (HDouss) [inline src/main/java/org/takes/facets/fork/MediaType.java]: @Happy-Neko This is a bad name. Please check: http://www.yegor256.com/2015/03/09/objects-end-with-er.html

File: src/main/java/org/takes/facets/fork/MediaType.java [hunk 2]
```diff
@@ -112,7 +118,7 @@ private static Double priority(final String text) {
         final String[] parts = MediaType.split(text);
         final Double priority;
         if (parts.length > 1) {
-            final String num = parts[1].replaceAll("[^0-9\\.]", "");
+            final String num = CLEANER.matcher(parts[1]).replaceAll("");
```
Comment (HDouss) [inline src/main/java/org/takes/facets/fork/MediaType.java]: @Happy-Neko You should access this field in a static way.

File: src/main/java/org/takes/misc/Href.java [hunk 3]
```diff
@@ -54,6 +55,11 @@
 public final class Href implements CharSequence {
 
     /**
+     * Trailer slash pattern.
+     */
+    private static final Pattern TRAILER = Pattern.compile("/$");
```
Comment (HDouss) [inline src/main/java/org/takes/misc/Href.java]: @Happy-Neko This is a bad name. Please check: http://www.yegor256.com/2015/03/09/objects-end-with-er.html

File: src/main/java/org/takes/misc/Href.java [hunk 4]
```diff
@@ -195,9 +201,11 @@ public String bare() {
     public Href path(final Object suffix) {
         return new Href(
             URI.create(
-                new StringBuilder(this.uri.toString().replaceAll("/$", ""))
-                    .append('/')
-                    .append(Href.encode(suffix.toString())).toString()
+                new StringBuilder(
+                    TRAILER.matcher(this.uri.toString()).replaceAll("")
```
Comment (HDouss) [inline src/main/java/org/takes/misc/Href.java]: @Happy-Neko You should access this field in a static way.

**Comment by HDouss** [comment]
@Happy-Neko It could be easier to get rid of the static fields. These "constants" are used once each. You could just inline them.

**Comment by Happy-Neko** [comment]
@HDouss constant names changed, thanks.

As for the static variable vs local I thinks the point of the ticket was to save time by compiling regexp only once. When inlined they will be compiled each time method is called, essentially the same as `String.replaceAll()`.

**Comment by HDouss** [comment]
@Happy-Neko Right ! Thanks.

**Comment by HDouss** [comment]
@rultor merge.

**Comment by yegor256** [comment]
@rultor merge

**Comment by davvd** [comment]
@elenavolokhova please, help us to evaluate quality here, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by elenavolokhova** [comment]
@HDouss Looks good!

**Comment by HDouss** [comment]
@davvd Could I be paid here please ?

**Comment by davvd** [comment]
> @davvd Could I be paid here please ?

@hdouss @yegor256 please help us here

**Comment by yegor256** [comment]
@davvd pls add extra 15 mins to @HDouss

**Comment by davvd** [comment]
> @davvd pls add extra 15 mins to @HDouss 

@yegor256 I've paid 15 mins to @hdouss in transaction `80470757`

**Comment by elenavolokhova** [comment]
@davvd @yegor256 What about payment for QA review? :)

**Comment by yegor256** [comment]
@elenavolokhova now I understand what happened. you didn't address your "looks good" message to the right person. see?

**Comment by elenavolokhova** [comment]
@yegor256 Oh, yes! :( Sorry about that... Sometimes QA can make a mistake too :)

**Comment by yegor256** [comment]
@davvd pls add extra 5 mins to @elenavolokhova

**Comment by yegor256** [comment]
@elenavolokhova yeah, it happens)

**Comment by davvd** [comment]
> @davvd pls add extra 5 mins to @elenavolokhova 

@yegor256 I've sent 5 mins extra to @elenavolokhova in transaction ID "80818349"

**Pull request merged**